### ISSUE TITLE
has anyone tried the USG-3p?

### ISSUE THREAD

**Issue by cdoublejj** [WireGuard/wireguard-vyatta-ubnt#128]
has anyone tried the usg-3p?

**Comment by jbeez** [comment]
Yes, I have a site2site from my parents usg3p to my pfsense box at my house. I haven't thoroughly tested it but seems fine so far.

**Comment by Looooopy** [comment]
i have trouble to reach 15Mbit/s but i have a 1000/1000 conntected to a site with 100/100.
So that limit the selection on what i can do.  

I have tried wireguard on My linux computer/ My linux server and now on unifi USG-3P router to rule out things but have no clue why its so slow.

I have measure with iperf3.

**Comment by jbeez** [comment]
> i have trouble to reach 15Mbit/s but i have a 1000/1000 conntected to a site with 100/100. So that limit the selection on what i can do.
> 
> I have tried wireguard on My linux computer/ My linux server and now on unifi USG-3P router to rule out things but have no clue why its so slow.
> 
> I have measure with iperf3.

What is your setup,
Cable modem/ONT/etc? What feeds into your usg3 for wan service?

On the USG3 do you have QoS/DPI/etc turned on? Those extraneous features can really bog down the performance on those units.

My parents have cable, I believe its 120/12 they get 123/12 on the unifi speed test which isn't even as accurate IMO as iperf, I know those units can perform better than 15mbps.

When you say you are measuring 15mbps, is that overall or only through a wireguard tunnel?

**Comment by cdoublejj** [comment]
it doesn't even matter you can't use wireguard if you host unifi off windows all the instruction are in Debian as a fuck you to windows users. So basically if your setup is windows server + unifi, go shell out for a Dream Machine Pro or SE. CGNAT fuckin sucks.

**Comment by cdoublejj** [comment]
it doesn't even matter you can't use wireguard if you host unifi off windows all the instruction are in Debian as a fuck you to windows users. So basically if your setup is windows server + unifi, go shell out for a Dream Machine Pro or SE. CGNAT fuckin sucks.

**Comment by cdoublejj** [comment]
Just to be clear the instructed commands are to be run on the unifi controller host or in an a SSH session on the gateway it's self?

**Comment by Looooopy** [comment]
I have fiber connection on my end 1GBit both ways on the other side we have fiber 100Mbit both ways.

I have disabled DPI no change.
I can't see the CPU go up much neither and no peak on memory on the USG.
I have measured internally throug the network behind the wireguard instace and i can max out a connection to 2.5 GBit on my end.

I have also tried 5G connection as external interface with exact same problem.

So there is something that must be problematic on the USG, because i have also tried port forwarding to an instance wireguard server to my own machine with the same result.

15Mbps is then i tunnel but when i do not tunnel i get my 1GBit fully usable.

**Comment by Looooopy** [comment]
> Just to be clear the instructed curl/debian commands in the instructions are to be run on the unifi controller host or in an a SSH session on the gateway it's self?

ssh on the gateway

**Comment by psonnosp** [comment]
Are there a guide for making this work somewhere? Can only find stuff regarding other devices.
